x-pack/plugin/sql/qa/src/main/java/org/elasticsearch/xpack/sql/qa/jdbc/JdbcAssert.java

```diff
@@ -269,11 +268,11 @@ else if (type == Types.DOUBLE) {
                     } else if (type == Types.FLOAT) {
                         assertEquals(msg, (float) expectedObject, (float) actualObject, lenientFloatingNumbers ? 1f : 0.0f);
                     } else if (type == Types.OTHER) {
-                        if (expectedObject instanceof Geometry && actualObject instanceof String) {
+                        if (actualObject instanceof org.elasticsearch.geo.geometry.Geometry) {
```

Comment:
Please import to avoid the qualifier.

Reply:
@bpintea can you please advice what type the ODBC driver indicates (if at all) for geo?

Reply:
> @bpintea can you please advice what type the ODBC driver indicates (if at all) for geo?
> 
> Own update: it looks like both geometry and geography are mapped to `-151`.

@costin There isn't any specification on how the geo types should be conveyed in ODBC. Most DBMSes use private types/extensions (like [MS SQL's "user defined" -151](https://docs.microsoft.com/en-us/sql/relational-databases/native-client/odbc/large-clr-user-defined-types-odbc)). WKT/B seems to have most support (also among some BI tools and Excel plug-ins), with GeoJSON picking up as well.

An idea would be to apply the same logic as with `KEYWORD`, `TEXT` and `IP`: give the new geography/geometry types respective names, but reuse a character (`12`) or the binary (`-3`) type.
Since Geo is not part of the ODBC standard, no 3rd party existing app could "out of the box" make use of whatever data types we'll implement. OTOH, the values encoding (WKT/B) could be used.